src/Symfony/Bundle/FrameworkBundle/EventListener/SuggestMissingPackageSubscriber.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\EventListener;
+
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleErrorEvent;
+use Symfony\Component\Console\Exception\NamespaceNotFoundException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Suggests a package, that should be installed (via composer),
+ * if the package is missing, and the input command namespace can be mapped to a Symfony bundle.
+ *
+ * @author Przemysław Bogusz <przemyslaw.bogusz@tubotax.pl>
+ */
+final class SuggestMissingPackageSubscriber implements EventSubscriberInterface
+{
+    const SUGGESTED_PACKAGES = [
+        'doctrine' => [
+            'Doctrine ORM',
+            'symfony/orm-pack',
+        ],
+        'generate' => [
+            'SensioGeneratorBundle',
+            'sensio/generator-bundle',
+        ],
+        'make' => [
+            'MakerBundle',
+            'symfony/maker-bundle --dev',
+        ],
+        'server' => [
+            'Symfony Web Server',
+            'symfony/web-server-bundle --dev',
+        ],
+    ];
+
+    public function onConsoleError(ConsoleErrorEvent $event)
+    {
+        if ($event->getError() instanceof NamespaceNotFoundException) {
+            if (null !== $event->getInput()->getFirstArgument()) {
+                $namespace = explode(':', $event->getInput()->getFirstArgument())[0];
+
+                if (isset(self::SUGGESTED_PACKAGES[$namespace])) {
+                    $suggestion = self::SUGGESTED_PACKAGES[$namespace];
+                    $message = sprintf("It seems that you are trying to run a command from '%s' bundle/package, but you do not have it installed. You may try installing the missing bundle/package by running:\n\n    composer require %s", $suggestion[0], $suggestion[1]);
+                    throw new NamespaceNotFoundException($message);
```

Comment:
`$event->setError()` instead of throwing a new Exception, this way other error event listeners can still be actioned.

Reply:
Done